e2e/test/RegistryManagerImportDevicesTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class RegistryManagerImportDevicesTests
+    {
+#pragma warning disable CA1823
+        private readonly TestLogging _log = TestLogging.GetInstance();
+
+        // A bug in either Storage or System.Diagnostics causes an exception during container creation
+        // so for now, we need to disable this.
+        // https://github.com/Azure/azure-sdk-for-net/issues/10476
+        //private readonly ConsoleEventListener _listener = TestConfig.StartEventListener();
+#pragma warning restore CA1823
+
+        private const string ImportFileNameDefault = "devices.txt";
+
+        private static readonly IReadOnlyList<JobStatus> s_incompleteJobs = new[]
+        {
+            JobStatus.Running,
+            JobStatus.Enqueued,
+            JobStatus.Queued,
+            JobStatus.Scheduled,
+            JobStatus.Unknown,
+        };
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task RegistryManager_ImportDevices()
+        {
+            StorageContainer storageContainer = null;
+            string deviceId = $"{nameof(RegistryManager_ImportDevices)}-{StorageContainer.GetRandomSuffix(4)}";
+            var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            _log.WriteLine($"Using deviceId {deviceId}");
+
+            try
+            {
+                // arrange
+
+                string containerName = StorageContainer.BuildContainerName(nameof(RegistryManager_ImportDevices));
+                storageContainer = await StorageContainer
+                    .GetInstanceAsync(containerName)
+                    .ConfigureAwait(false);
+                _log.WriteLine($"Using container {storageContainer.Uri}");
+
+                Stream devicesFile = ImportExportDevicesHelpers.BuildDevicesStream(
+                    new List<ExportImportDevice>
+                    {
+                        new ExportImportDevice(
+                            new Device(deviceId)
+                            {
+                                Authentication = new AuthenticationMechanism { Type = AuthenticationType.Sas }
```

Comment:
Do you want to parameterize this test on AuthenticationType in preparation for the new authentication type?